js/source/worker.js

```diff
@@ -130,13 +130,28 @@ util.extend(Worker.prototype, {
      * function taking `(name, workerSourceObject)`.
      *  @private
      */
-    'load worker source': function(params, callback) {
+    'load worker source': function(map, params, callback) {
         try {
             this.self.importScripts(params.url);
             callback();
         } catch (e) {
             callback(e);
         }
+    },
+
+    getWorkerSource: function(mapId, type) {
```

Comment:
They are keyed by (mapId, type) pairs.  They're keyed by `mapId` so that WorkerSource code wouldn't need to change (i.e., each WorkerSource instance is still tied to a specific map -- see bullet 3 in "Specific Notes").  They're keyed by type because each `WorkerSource` handles requests from multiple individual sources (note that this has been true since before the custom source refactor)